packages/fx-core/src/component/generator/generateAction.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import AdmZip from "adm-zip";
+import path from "path";
+import { GenerateContext } from "./generateContext";
+import { fetchZipUrl, templateZipName, unzip } from "./utils";
+import fs from "fs-extra";
+import { fetchZipFromUrl } from "../../common/template-utils/templatesUtils";
+import {
+  defaultTimeoutInMs,
+  defaultTryLimits,
+  sampleDownloadBaseUrl,
+  templateDownloadBaseUrl,
+} from "./constant";
+import { getTemplatesFolder } from "../../folder";
+import { MissKeyError } from "./error";
+
+export interface GenerateAction {
+  name: string;
+  run: (context: GenerateContext) => Promise<void>;
+}
+
+export enum GenerateActionName {
```

Comment:
I don't get why do we rewrite actions? Can we just re-use the template utils?

Reply:
There are some small differences between template utils actions and new actions, so I think it's better to rewrite them.